azurelinuxagent/ga/exthandlers.py

```diff
@@ -514,6 +514,7 @@ def handle_ext_handlers(self, etag=None):
                 # For SC extensions, overwrite the HandlerStatus with the relevant message
                 else:
                     handler_i.set_handler_status(message=depends_on_err_msg, code=-1)
+                    ext_handler.dependency_failed = True
```

Comment:
Sorry to do this again, but there is one major problem with this approach. The `ext_handler` state is not maintained after one iteration. The drawback of this is that we would report the status of the `dependency_failed` only once to CRP instead of reporting it consistently until a new GS.